### DISCUSSION TITLE
Context managers are cool... and I forgot you can "combine" them to avoid too much nesting

### DISCUSSION THREAD

**Discussion by villares** [py5coding/py5generator#444] [Show and tell]
```
...
with push_matrix(), begin_closed_shape():
    scale(s)  # scale factor
    vertices(p_list)
```

**Comment by hx2A** [comment]
Neat, I didn't know you could do that with context managers. Thanks for posting this and sharing.

Also, thanks for the encouragement to add context managers to py5! Remember, you told me it was a useful feature in processing.py that would add value to py5. Great idea, as always!
